src/rantanplan/core.py

```diff
@@ -783,3 +815,20 @@ def has_single_liaisons(liaisons):
     :rtype: bool
     """
     return not any(i == j == 1 for i, j in zip(liaisons, liaisons[1:]))
+
+
+def remove_exact_length_matches(lines):
+    """
+    Removes key "length_range" on lines with an exact length match
+    :param lines: List of dictionary lines of the poem
+    :return: Returns the lines list without  the "length_range" on lines with an exact length match
+    """
+    for line in lines:
+        rhythm_dict = {}
+        for key, value in line["rhythm"].items():
+            if key != "length_range":
+                rhythm_dict.update({key: value})
+            elif value['min_length'] != value['max_length']:
+                rhythm_dict.update({key: value})
+        line["rhythm"] = rhythm_dict
+    return lines
```

Comment:
Maybe something like this? Not even sure the first `if` is necessary since `"length_range"` is always present, IFAIU.
```suggestion
        if "length_range" in line["rhythm"]:
            ranges = line["rhythm"]["length_range"]
            # ranges["min_length"] is always line["rhythm"]["length"]
            if ranges["min_length"] == ranges["max_length"]:
                del line["rhythm"]["length_range"]
    return lines
```

Reply:
I'll check that

Reply:
ranges["min_length"] is not always line["rhythm"]["length"] bacuse if it finds a structure and recalculates, it can have a higher value